mlflow/store/artifact/azure_blob_artifact_repo.py

```diff
@@ -93,7 +93,7 @@ def list_artifacts(self, path=None):
         if path:
             dest_path = posixpath.join(dest_path, path)
         infos = []
-        prefix = dest_path + "/"
+        prefix = dest_path if dest_path.endswith("/") else dest_path + "/"
```

Comment:
Should we add a test where we call `list_artifacts` with `some_path` and `some_path/` and assert that `walk_blobs` is called with `some_path/` in both cases?

Reply:
Done. Thanks for the suggestion!